### PR TITLE
Add P-chain `GetBlockByHeight` API method

### PR DESCRIPTION
**Pull request by dhrubabasu**
## Why this should be merged

Adds `platformvm.getBlockByHeight`

## How this works

pretty straightforward

## How this was tested

UT and locally

### CODE REVIEW

File: vms/platformvm/service.go
Comment (ceyonur) [inline vms/platformvm/service.go:2617]: this is not somehow possible to be triggered by clients right? Like by passing wrong heights?

File: vms/platformvm/service.go
Comment (ceyonur) [inline vms/platformvm/service.go:2629]:
```suggestion
		return fmt.Errorf("couldn't encode block with id %s as string: %w", blockID, err)
```

**Review by abi87 — COMMENTED** [review]
I don't see issues apart from what @ceyonur mentioned. Also this is definitely going to help dowmstream!
I wonder is we need something similar at the proposerVM level?

File: vms/platformvm/service.go
Comment (dhrubabasu) [reply vms/platformvm/service.go:2629]: I copied this error message from the AVM to be consistent:
https://github.com/ava-labs/avalanchego/blob/master/vms/avm/service.go#L101

Will defer to @StephenButtolph

File: vms/platformvm/service.go
Comment (dhrubabasu) [reply vms/platformvm/service.go:2617]: If they pass the wrong height, it would exit at line 2610

**Comment by dhrubabasu** [comment]
> I don't see issues apart from what @ceyonur mentioned. Also this is definitely going to help dowmstream! I wonder is we need something similar at the proposerVM level?

The idea is to add it `GetBlockID` to the `block.ChainVM` interface for use in bootstrapping. We will need to implement that method in the ProposerVM to maintain compliance.

File: vms/platformvm/client.go
Comment (gyuho) [inline vms/platformvm/client.go:259]: Also, asked in the previous PR. Do we ever need to support multiple heights as input?

**Comment by gyuho** [comment]
c.f., https://github.com/ava-labs/avalanchego/pull/1447

File: vms/platformvm/client.go
Comment (dhrubabasu) [reply vms/platformvm/client.go:259]: Unlikely since it's not in the X-chain API set. Outside of the scope of this PR

**Review by abi87 — APPROVED** [review]
lgtm, pending nits from other reviewers

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**
